Content.Server/Atmos/Reactions/PlasmaFireReaction.cs

```diff
@@ -9,78 +10,81 @@ namespace Content.Server.Atmos.Reactions
+            AtmosphereSystem atmosphereSystem,
+            float heatScale)
         {
-            var energyReleased = 0f;
-            var oldHeatCapacity = atmosphereSystem.GetHeatCapacity(mixture, true);
             var temperature = mixture.Temperature;
-            var location = holder as TileAtmosphere;
             mixture.ReactionResults[(byte)GasReaction.Fire] = 0;
 
-            // More plasma released at higher temperatures.
-            var temperatureScale = 0f;
-
-            if (temperature > Atmospherics.PlasmaUpperTemperature)
-                temperatureScale = 1f;
-            else
-            {
-                temperatureScale = (temperature - Atmospherics.PlasmaMinimumBurnTemperature) /
-                                   (Atmospherics.PlasmaUpperTemperature - Atmospherics.PlasmaMinimumBurnTemperature);
-            }
-
-            if (temperatureScale > 0)
+            // React faster and with less oxygen per plasma at higher temperatures, up to a maximum.
+            var temperatureReactionScale = ProportionOfRange(temperature,
+                Atmospherics.PlasmaMinimumBurnTemperature,
+                Atmospherics.PlasmaUpperTemperature);
+            if (temperatureReactionScale > 0)
             {
-                var oxygenBurnRate = Atmospherics.OxygenBurnRateBase - temperatureScale;
-                var plasmaBurnRate = 0f;
-
                 var initialOxygenMoles = mixture.GetMoles(Gas.Oxygen);
                 var initialPlasmaMoles = mixture.GetMoles(Gas.Plasma);
 
-                // Supersaturation makes tritium.
-                var oxyRatio = initialOxygenMoles / initialPlasmaMoles;
-                // Efficiency of reaction decreases from 1% Plasma to 3% plasma:
-                var supersaturation = Math.Clamp((oxyRatio - Atmospherics.SuperSaturationEnds) /
-                                                 (Atmospherics.SuperSaturationThreshold -
-                                                  Atmospherics.SuperSaturationEnds), 0.0f, 1.0f);
-
-                if (initialOxygenMoles > initialPlasmaMoles * Atmospherics.PlasmaOxygenFullburn)
-                    plasmaBurnRate = initialPlasmaMoles * temperatureScale / Atmospherics.PlasmaBurnRateDelta;
-                else
-                    plasmaBurnRate = temperatureScale * (initialOxygenMoles / Atmospherics.PlasmaOxygenFullburn) / Atmospherics.PlasmaBurnRateDelta;
+                // The maximum rate based on the relative proportions of reactants, expressed as moles of plasma burned.
+                var maximumReactionRate =
+                    Math.Min(initialPlasmaMoles, initialOxygenMoles / Atmospherics.PlasmaOxygenFullburn);
+                var reactionRate = maximumReactionRate * temperatureReactionScale / Atmospherics.PlasmaBurnRateDelta;
 
-                if (plasmaBurnRate > Atmospherics.MinimumHeatCapacity)
+                // TODO I have no fucking idea why the rate is compared against a heat capacity here.
```

Comment:
Maybe because rate times an implicit tickRate is a change in heat capacity?

Reply:
Can you find out if the rate is in unit/tick or unit/sec?

Reply:
As far as I can tell, it's units of gas per [AtmosTick](https://github.com/space-wizards/space-station-14/blob/68ce53ae17985876d6d112b764b2144964a9f42e/Content.Server/Atmos/EntitySystems/AtmosphereSystem.CVars.cs#L26).

`Atmospherics.MinimumHeatCapacity` is only compared to other heat capacities or used as a default heat capacity except in this case, and in the case of the Freezon + N2 reaction, which seems to be very similar to the plasmox reaction, so it may be a copy/paste.

So if `reactionRate`'s dimension `mass / time` and `Atmospherics.MinimumHeatCapacity`'s is (real world) `energy / (mass * temperature)`, it reaaaaally seems to me like comparing them does not describe anything physically meaningful.